linked_list/main.py

```diff
@@ -62,34 +68,166 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        self.elements = []      
+        if (elements == None or len(elements) <1):
+            #self.start, self.end = Node(None, None), Node(None, None)
+            self.start, self.end = None, None
+        if elements != None and len(elements) > 0:
+            self.elements = elements
+            index = 1
+            for e in range(len(self.elements)):
+                try:
+                    e = Node(e, Node(self.elements[index]))
+                except IndexError:
+                    e = Node(e)
+            try:
+                self.start = Node(self.elements[0], self.elements[1])
+                self.end = Node(self.elements[-1])
+            except IndexError:
+                self.start = Node(self.elements[0])
+                self.end = self.start
+
+        
+        
+    """
+        self.elements = elements            #a lis
+        if (self.elements == None or len(self.elements) <1):
+            #self.start, self.end = Node(None, None), Node(None, None)
+            self.start, self.end = None, None
+        if self.elements != None and len(self.elements) > 0:
+            index = 1
+            for e in range(len(self.elements)):
+                try:
+                    e = Node(e, self.elements[index])
+                except IndexError:
+                    e = Node(e)
+            self.start = self.elements[0]
+            self.end = self.elements[-1]
+    """
+        
+    """
+        end_index = 1
+        if self.elements != None:
+            if len(self.elements) >= 1:
+                self.start = self.elements[0]
+                self.end = self.elements[-1]
+                while end_index <= (len(self.elements)+1):
+                    for i in elements:
+                        try:
+                        #if i != self.elements:
+                            i = Node(i, self.elements[end_index])
+                            end_index += 1
+                            #print(end_index)
+                        except IndexError:
+                            i = Node(i)
+                            break
+                    break
+    """
+    """
+        if len(self.elements) > 1:
+            self.start = Node(self.elements[0], Node(self.elements[1]))             #Node pbject
+            self.end = Node(self.elements[-1])
+            
+        if len(self.elements) == 1:
+            self.start = Node(self.elements[0])             #Node pbject
+            self.end = Node(self.elements[-1])
+            
+    """
 
-    def __str__(self):
-        pass
 
+    def __str__(self):
+        if self.elements == None:
+            return "[]"
+        return "%s" % (self.elements)
+        
     def __len__(self):
-        pass
+        return len(self.elements)
 
     def __iter__(self):
-        pass
-
+        return self.elements
```

Comment:
If you just had all the nodes separately (instead of putting them in a list) you could make them iterable by traversing through them one by one (begin with the start Node and then follow the .next pointer to the second Node, etc). For example you could have a generator that would yield each Node it comes to as it traverses from the start Node to the end Node.